### PR TITLE
enhancement: add local-fuel-node arg to spin one up

### LINKED ISSUE #561: Allow test Fuel node to be started in indexer binary
- This was a piece of feedback that came with the https://github.com/FuelLabs/fuel-indexer/issues/544 suggestion
- We should add the ability to run a local Fuel node from the main `fuel-indexer` service
- We actually used to have this functionality [way back when](https://github.com/FuelLabs/fuel-indexer/blob/cb7602c91fbca734e7a7253b3b5f376d60f85a12/indexer/src/bin/service.rs#L17)
- Ideally I'd include some flag `cargo run --bin fuel-indexer -- --local-fuel-node` and it would spin up my node using the default settings
- This would only apply to `fuel-indexer`, not `fuel-indexer-api-server`

### PR DESCRIPTION
**Pull request by rvmelkonian**
Closes #561.

## Changelog
• Adds a new `local_fuel_node` to the `IndexerArgs struct`.
• Updates `fuel-indexer.rs` to run the node if the flag is present.
• Brings in `fuel-core v0.17.0` to our Cargo.toml

## Testing plan 
I have done manual tests. 
•~~When `cargo run --bin fuel-indexer` is ran, the indexer starts, with no local fuel node~~.
•~~When `cargo run --bin fuel-indexer -- --local-fuel-node` is ran, the indexer starts and the local fuel node with default settings starts up~~. 

## Notes
I didn't write any unit-tests or integration tests here as it didn't seem necessary, but happy to if that's preferred. 
The PR looks large due to the `Cargo.lock` change.

### CODE REVIEW

**Review by ra0x3 — CHANGES_REQUESTED** [review]
@rvmelkonian 
- It works! 
- Some small feedback related to default settings and feature flagging

File: packages/fuel-indexer/Cargo.toml [hunk 1]
```diff
@@ -15,6 +15,7 @@ bincode = "1.3.3"
 cfg-if = "1.0"
 chrono = { version = "0.4", features = ["serde"] }
 clap = { version = "3.1", features = ["cargo", "derive", "env"] }
+fuel-core = { version = "0.17.0" }
```
[the remark this answers is not in the document]
Comment (rvmelkonian) [reply packages/fuel-indexer/Cargo.toml:18]: Yes. I was worried about increasing the size of the binary also. Good idea, let's feature flag it.

File: packages/fuel-indexer/src/bin/fuel-indexer.rs [hunk 2]
```diff
@@ -55,6 +56,13 @@ pub async fn main() -> Result<()> {
 
     let mut service = IndexerService::new(config.clone(), pool.clone(), rx).await?;
 
+    let _local_node = if opt.local_fuel_node{
+        let s = FuelService::new_node(Config::local_node()).await.unwrap();
```
[the remark this answers is not in the document]
Comment (deekerno) [reply packages/fuel-indexer/src/bin/fuel-indexer.rs:60]: You _just_ beat me to it! I think that using port 4000 would be best as the indexer expects that by default. So you could one of two things:
- Add a function to generate a configuration struct for the local node where the only change is the port
- Take whatever port the OS hands to `fuel-core` and let the indexer service know about it

I'll let you choose what you want to do. 👌🏽

**Comment by rvmelkonian** [comment]
The command to run now is `cargo run --bin fuel-indexer --features local-node` 

The `FuelService` is assigned port 4000.

Because this fuel-core is an external dependancy and not within our workspace, the compiler complained abou missing types when I used the macro `if cfg!()`, so I used the cfg attribute with a scoped block instead.

**Comment by deekerno** [comment]
```
~/dev/fuel/indexer rich/561-local-node
❯ ps ax | rg fuel-indexer
95660 s004  S+     0:01.42 target/debug/fuel-indexer

~/dev/fuel/indexer rich/561-local-node
❯ lsof -ti:4000
95660
```

Works for me! I'll let Rashad have the final approval as he was the one to request changes.

**Review by ra0x3 — APPROVED** [review]
Nice!

**Pull request merged**